aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -256,6 +270,43 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    public entry fun offer_rotation_capability(account: &signer, rotation_capability_signature: vector<u8>, public_key: vector<u8>, intended_receiver_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(intended_receiver_address), error::not_found(EACCOUNT));
+        assert!(
+            vector::length(&public_key) == 32,
+            error::invalid_argument(EMALFORMED_PUBLIC_KEY)
+        );
+        assert!(
+            vector::length(&rotation_capability_signature) == 64,
+            error::invalid_argument(EMALFORMED_PROOF_OF_KNOWLEDGE)
+        );
+
+        let account_resource = borrow_global_mut<Account>(addr);
+
+        let rotation_capability_offer_proof = RotationCapabilityOfferProof {
+            sequence_number: account_resource.sequence_number,
+            intended_receiver_address,
+        };
+
```

Comment:
you need to verify that the public key is the same as the auth key
also note that this only works with the ed25519 key, so we probably want to call this offer_rotation_capability_ed25519